bluemira/builders/EUDEMO/vacuum_vessel.py

```diff
@@ -0,0 +1,175 @@
+from bluemira.geometry.placement import BluemiraPlacement
+from bluemira.geometry.tools import offset_wire, revolve_shape, slice_shape
+from bluemira.geometry.wire import BluemiraWire
+
+
+class VacuumVesselBuilder(Builder):
+    """
+    Builder for the vacuum vessel
+    """
+
+    _required_params: List[str] = [
+        "r_vv_ib_in",
+        "r_vv_ob_in",
+        "tk_vv_in",
+        "tk_vv_out",
+        "g_vv_bb",
+        "n_TF",
+    ]
+    _params: Configuration
+    _ivc_koz: BluemiraWire
+
+    def __init__(
+        self,
+        params,
+        build_config: BuildConfig,
+        ivc_koz: BluemiraWire,
+    ):
+        super().__init__(
+            params,
+            build_config,
+            ivc_koz=ivc_koz,
+        )
+
+    def reinitialise(self, params, ivc_koz) -> None:
+        """
+        Reinitialise the parameters and boundary.
+
+        Parameters
+        ----------
+        params: dict
+            The new parameter values to initialise this builder against.
+        """
+        super().reinitialise(params)
+
+        self._ivc_koz = ivc_koz
+
+    def build(self) -> Component:
+        """
+        Build the vacuum vessel component.
+
+        Returns
+        -------
+        component: Component
+            The Component built by this builder.
+        """
+        super().build()
+
+        component = Component(name=self.name)
+        component.add_child(self.build_xz())
+        component.add_child(self.build_xy())
+        component.add_child(self.build_xyz())
+        return component
+
+    def build_xz(self):
+        """
+        Build the x-z components of the vacuum vessel.
+        """
+        inner_vv = offset_wire(
+            self._ivc_koz, self._params.g_vv_bb.value, join="arc", open_wire=False
+        )
+        angle_1 = 80
+        angle_2 = 160
+        outer_vv = varied_offset(
+            inner_vv,
+            self._params.tk_vv_in.value,
+            self._params.tk_vv_out.value,
+            angle_1,
+            angle_2,
+            num_points=300,
+        )
+        face = BluemiraFace([outer_vv, inner_vv])
+        self._vv_face = face
+
+        body = PhysicalComponent("body", face)
+        body.plot_options.face_options["color"] = BLUE_PALETTE["VV"][0]
+
+        component = Component("xz", children=[body])
+        bm_plot_tools.set_component_view(component, "xz")
+        return component
+
+    def build_xy(self):
+        """
+        Build the x-y components of the vacuum vessel.
+        """
+        xy_plane = BluemiraPlacement.from_3_points([0, 0, 0], [1, 0, 0], [1, 1, 0])
+        intersections = slice_shape(self._vv_face.boundary[0], xy_plane)
+        print(intersections)
+        r_inters = sorted(intersections[:, 0])
+        r_ib_out, r_ob_out = r_inters
+        intersections = slice_shape(self._vv_face.boundary[1], xy_plane)
+        r_inters = sorted(intersections[:, 0])
+        r_ib_in, r_ob_in = r_inters
```

Comment:
I think a well named function would improve the readability of this segment of code. Maybe something like

```python
r_ib_out, r_ob_out = _find_intersections_x_coord(self._vv_face.boundary[0], xy_plane)
r_ib_in, r_ob_in = _find_intersections_x_coord(self._vv_face.boundary[1], xy_plane)
```
With the named function, I immediately know that the in/out radii are the x-coordinates of the intersections of the boundaries and the xy-plane. And I also immediately know that both sets of radii are computed in the same way. In 2 lines rather than 6, and all without having to read and understand the code that's doing it.